android/demos/compose/src/main/kotlin/com/uber/rib/compose/root/main/MainRouter.kt

```diff
@@ -27,48 +27,56 @@ import com.uber.rib.core.BasicViewRouter
 class MainRouter(
   view: ComposeView,
   interactor: MainInteractor,
+  private val parentView: ViewGroup,
   private val scope: MainScope,
-  private val childContent: MainRouter.ChildContent
+  private val childContent: ChildContent
 ) : BasicViewRouter<ComposeView, MainInteractor>(view, interactor) {
 
   private var loggedOutRouter: LoggedOutRouter? = null
   private var loggedInRouter: LoggedInRouter? = null
 
+  override fun willAttach() {
+    super.willAttach()
+    parentView.addView(view)
+  }
+
+  override fun willDetach() {
+    super.willDetach()
+    parentView.removeView(view)
+  }
+
   internal fun attachLoggedOut() {
     if (loggedOutRouter == null) {
-      loggedOutRouter = scope.loggedOutScope(view).router().also {
+      loggedOutRouter = scope.loggedOutScope(childContent.fullScreenSlot).router().also {
         attachChild(it)
-        childContent.fullScreenContent = it.presenter.composable
       }
     }
   }
 
   internal fun attachLoggedIn(authInfo: AuthInfo) {
     if (loggedInRouter == null) {
-      loggedInRouter = scope.loggedInScope(view, authInfo).router().also {
-        attachChild(it)
-        childContent.fullScreenContent = it.presenter.composable
-      }
+      loggedInRouter =
+        scope.loggedInScope(childContent.fullScreenSlot, authInfo).router().also {
+          attachChild(it)
+        }
     }
   }
 
   internal fun detachLoggedOut() {
     loggedOutRouter?.let {
-      childContent.fullScreenContent = null
       detachChild(it)
     }
     loggedOutRouter = null
   }
 
   internal fun detachLoggedIn() {
     loggedInRouter?.let {
-      childContent.fullScreenContent = null
       detachChild(it)
     }
     loggedInRouter = null
   }
 
   class ChildContent {
-    internal var fullScreenContent: (@Composable () -> Unit)? by mutableStateOf(null)
+    var fullScreenSlot: MutableState<(@Composable () -> Unit)> = mutableStateOf({})
```

Comment:
since our slots are non-nullable, can you please revert back to the delegate usage? (no MutableState wrapper)